feat(opensearch): add support for latest amazon opensearch service 1.3

prashanttct07 opened the pull request:
fixes #21414 
Description:
AWS Launched the support for Amazon OpenSearch Service 1.3 as per the announcement here: https://aws.amazon.com/about-aws/whats-new/2022/07/amazon-opensearch-service-supports-version-1-3/

AWS CDK does not support the latest version 1.3 yet, and this PR adds the support of Amazon OpenSearch Service 1.3 in CDK. Many people uses CDK to launch various AWS services and OpenSearch is one of them and having updated with latest version will help me to launch the Amazon OpenSearch Service using CDK.

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 3d513564b063c5a4bd2cd9f7c0577cb560e70ebf
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4d61ed73-01ad-40fb-8651-e590dece98ef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra reviewed (dismissed):
Please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request)).

Additionally, make sure that your PR body describes the problem the PR is solving, and the design approach and alternatives considered. Explain why the PR solves the problem. A link to an issue is helpful, but does not replace an explanation of your thought process.

More specific to the code change, I see that OpenSearch also has 2.0 and 2.1. Can you add those as well?

prashanttct07 commented:
I have updated the title as per the guideline. And updated the description. 

Also, this is specific to launching an Amazon OpenSearch Service (managed service by AWS) using CDK which supports a maximum version upto 1.3. 

OpenSearch 2.0 and 2.1 is there as part of OpenSearch Project (Open Source) but not there in service. We can add support of newer version in CDK as and when it becomes available with Amazon OpenSearch Service. 

Let me know if anything else is needed for same.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 4876fe436a41a03d2762a01671b0f2c0acabbb41
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f0251a34-9fc8-4e4e-94d1-a0443521a5c5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
